### PR TITLE
Add HTTP statuses

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ❌
| Tests pass?   | -
| Fixed issues  | -

### CODE REVIEW

File: src/Status.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    public const INSUFFICIENT_STORAGE = 507;
+    public const LOOP_DETECTED = 508;
+    public const BANDWITH_LIMIT_EXCEEDED = 509;
+    public const NOT_EXTENDED = 510;
+    public const NETWORK_AUTHENTICATION_REQUIRED = 511;
+
+    /**
+     * @var array list of HTTP status texts
+     */
+    public static $statusTexts = [
```
Comment (rob006) [inline src/Status.php]: This should be a constant.

File: src/Status.php [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    public const MULTIPLE_CHOICES = 300;
+    public const MOVED_PERMANENTLY = 301;
+    public const FOUND = 302;
+    public const SEE_OTHER = 303;
+    public const NOT_MODIFIED = 304;
+    public const USE_PROXY = 305;
+    public const RESERVED = 306;
+    public const TEMPORARY_REDIRECT = 307;
+    public const PERMANENT_REDIRECT = 308;
+    public const TOO_MANY_REDIRECT = 310;
```
Comment (samdark) [inline src/Status.php]: Is this one standard? Haven't seen it previously anywhere.

File: src/Status.php [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    public const NOT_FOUND = 404;
+    public const METHOD_NOT_ALLOWED = 405;
+    public const NOT_ACCEPTABLE = 406;
+    public const PROXY_AUTHENTICATION_REQUIRED = 407;
+    public const REQUEST_TIMEOUT = 408;
+    public const CONFLICT = 409;
+    public const GONE = 410;
+    public const LENGTH_REQUIRED = 411;
+    public const PRECONDITION_FAILED = 412;
+    public const REQUEST_ENTITY_TOO_LARGE = 413;
```
Comment (samdark) [inline src/Status.php]: Isn't thins one called "Payload Too Large" in the spec? https://tools.ietf.org/html/rfc7231#section-6.5.11

File: src/Status.php [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+/**
+ * HTTP status codes
+ */
+final class Status
+{
+    public const CONTINUE = 100;
+    public const SWITCHING_PROTOCOLS = 101;
+    public const PROCESSING = 102;
+    public const CONNECTION_TIMEOUT = 118;
```
Comment (samdark) [inline src/Status.php]: Is it part of any RFC or other specification?

File: src/Status.php [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    public const METHOD_NOT_ALLOWED = 405;
+    public const NOT_ACCEPTABLE = 406;
+    public const PROXY_AUTHENTICATION_REQUIRED = 407;
+    public const REQUEST_TIMEOUT = 408;
+    public const CONFLICT = 409;
+    public const GONE = 410;
+    public const LENGTH_REQUIRED = 411;
+    public const PRECONDITION_FAILED = 412;
+    public const REQUEST_ENTITY_TOO_LARGE = 413;
+    public const REQUEST_URI_TOO_LONG = 414;
```
Comment (samdark) [inline src/Status.php]: This one is called "URI Too Long" in the RFC.

**Comment by samdark** [comment]
I like the idea and have a proposal for making it even more useful. How about adding phpdoc for each status:

1. Short description. We can get it from https://developer.mozilla.org/en-US/docs/Web/HTTP/Status
2. A link to RFC where the status code is defined.

File: src/Status.php (hunk 4 above)
Comment (rustamwin) [reply src/Status.php]: It was ported from Yii2  ¯\_(ツ)_/¯

Comment (samdark) [reply src/Status.php]: That doesn't guarantee it's correct.

**Comment by samdark** [comment]
@rustamwin going to add descriptions for the rest of the statuses?

**Comment by rustamwin** [comment]
Yes.

**Comment by rustamwin** [comment]
There is some statuses doesn't have description

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**